### ISSUE TITLE
mstile causing bundle duplication

### ISSUE THREAD

**Issue by tristan-f-r** [parcel-bundler/parcel#8821]
# 🐛 bug report

## 🎛 Configuration (.babelrc, package.json, cli command)

No special configuration options, just the parcel@2 logo from the parcel website and an HTML file:
```html
<head>
  <meta name="msapplication-TileColor" content="#FFFFFF" />
    <meta name="msapplication-TileImage" content="parcel.png?width=144&height=144" />
    <meta name="msapplication-square70x70logo" content="parcel.png?width=70&height=70" />
    <meta name="msapplication-square150x150logo" content="parcel.png?width=150&height=150" />
    <meta name="msapplication-square310x310logo" content="parcel.png?width=310&height=310" />
</head>
```

## 🤔 Expected Behavior

mstile should work as expected and not cause an error

## 😯 Current Behavior

```
Error: Bundles must have unique names

  AssertionError [ERR_ASSERTION]: Bundles must have unique names
      at BundlerRunner.nameBundles (/workspace/parcel-triage-mstile/node_modules/@parcel/core/lib/requests/BundleGraphRequest.js:390:23)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async BundlerRunner.bundle (/workspace/parcel-triage-mstile/node_modules/@parcel/core/lib/requests/BundleGraphRequest.js:341:7)
      at async Object.run (/workspace/parcel-triage-mstile/node_modules/@parcel/core/lib/requests/BundleGraphRequest.js:156:17)
      at async RequestTracker.runRequest (/workspace/parcel-triage-mstile/node_modules/@parcel/core/lib/RequestTracker.js:756:20)
      at async Object.run (/workspace/parcel-triage-mstile/node_modules/@parcel/core/lib/requests/ParcelBuildRequest.js:56:7)
      at async RequestTracker.runRequest (/workspace/parcel-triage-mstile/node_modules/@parcel/core/lib/RequestTracker.js:756:20)
      at async Parcel._build (/workspace/parcel-triage-mstile/node_modules/@parcel/core/lib/Parcel.js:400:11)
      at async Parcel.run (/workspace/parcel-triage-mstile/node_modules/@parcel/core/lib/Parcel.js:276:18)
      at async run (/workspace/parcel-triage-mstile/node_modules/parcel/lib/cli.js:376:7)
```

## 🔦 Context

I would like to allow my site to be added as a microsoft application, and that requires mstile metadata

## 💻 Code Sample

https://github.com/LeoDog896/parcel-triage-mstile

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.8.3
| Node             | 16.19.0
| npm/Yarn         | 8.19.3 / 1.22.19
| Operating System | Linux

**Comment by mischnic** [comment]
This is the relevant code: https://github.com/parcel-bundler/parcel/blob/adb01fe60c10dc119cffcd025f8c30a332547793/packages/transformers/html/src/dependencies.js#L120-L141

Not entirely sure why we have `needsStableName` here (which means files without hashes), is that needed for opengraph? 
This `needsStableName: true` also the reason why it tries to create multiple `parcel.png` bundles (as they all reference the same asset).

**Comment by tristan-f-r** [comment]
Related to #6702 which was patched by #6724
